render/render.js

```diff
@@ -489,18 +489,18 @@ module.exports = function($window) {
 	}
 
 	//lifecycle
-	function initLifecycle(source, vnode, hooks) {
-		if (typeof source.oninit === "function") source.oninit.call(vnode.state, vnode)
-		if (typeof source.oncreate === "function") hooks.push(source.oncreate.bind(vnode.state, vnode))
+	function initLifecycle(vnode, source, hooks) {
```

Comment:
Could you pull `vnode[source]` into its own variable? IMHO it's clearer. Also, use `prop`, not `source` for the property type, so it's clearer that it's the property, not the actual source in question.